usersync/ejector.go

```diff
@@ -13,10 +13,25 @@ type Ejector interface {
 
 type OldestEjector struct {
 	nonPriorityKeys []string
+	PriorityGroups  [][]string
 	FallbackEjector FallbackEjector
 }
 
+type PriorityBidderEjector struct {
+	PriorityGroups  [][]string
+	SyncerKey       string
+	OldestEjector   OldestEjector
+	FallbackEjector FallbackEjector
```

Comment:
Shouldn't the FallbackEjector be of type Ejector (interface)?

Reply:
I treated the `FallbackEjector` like the other `Ejectors` in the sense that it has it's own specific `Choose()` method that it implements, therefore I think it should be of type `FallbackEjector`

Reply:
I still don't understand why there is a `OldestEjector` field here. The `OldestEjector` has no state, so is not a variable. I could understand some sort of interface here in the `FallbackEjector` case, as an interface gives you the option to change what "fallback" logic is being used. But the `OldestEjector` field here just provides a roundabout way of referencing the function which is currently used as the `Choose()` method of the empty `OldestEjector` object. Why not have a static function and not spend a few bytes referencing a static, empty object?

Reply:
This was discussed in team time, but just wanted to comment here what was discussed. 

This is my fault I didn't update the PR for this part to make sense. I just pushed a commit where the `PriorityEjector` will take in a `TieEjector` object type `Ejector`. So with it now accepting an interface, in the future if we wanted to change how we handle ties in the ejector logic, we can change what ejector we pass in here. So right now, the `PriorityEjector` will take in the `OldestEjector` as it's Tie handler, so that's why the `OldestEjector` exists rather than a static function.